autoLogging/src/main/java/ru/otus/homework/LogAspect.java

```diff
@@ -0,0 +1,65 @@
+package ru.otus.homework;
+
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.stream.Collectors;
+
+@SuppressWarnings("ALL")
+@Slf4j
+public class LogAspect {
+
+    static Loggable createLoggedProxy() {
+        InvocationHandler ih = new LoggerHandler(new LoggableImpl());
+        return (Loggable) Proxy.newProxyInstance(LoggableImpl.class.getClassLoader(), new Class<?>[]{Loggable.class}, ih);
+    }
+
+    @FieldDefaults(level = AccessLevel.PRIVATE)
+    static class LoggerHandler implements InvocationHandler {
+        Loggable className;
+        HashSet<String> annotatedMethods = new HashSet<>();
```

Comment:
Лучше тип объявлять интерфейсом

```suggestion
        Set<String> annotatedMethods = new HashSet<>();
```